modules/gapi/src/backends/cpu/gcpuvideo.cpp

```diff
@@ -107,13 +107,74 @@ GAPI_OCV_KERNEL_ST(GCPUBackgroundSubtractor,
     }
 };
 
+GAPI_OCV_KERNEL_ST(GCPUKalmanFilter, cv::gapi::video::GKalmanFilter, cv::KalmanFilter)
+{
+    static void setup(const cv::GMatDesc&, const cv::GOpaqueDesc&,
+                      const cv::GMatDesc&, const cv::gapi::KalmanParams& kfParams,
+                      std::shared_ptr<cv::KalmanFilter> &state, const cv::GCompileArgs&)
+    {
+        state = std::make_shared<cv::KalmanFilter>(kfParams.transitionMatrix.rows, kfParams.measurementMatrix.rows,
+                                                   kfParams.controlMatrix.cols, kfParams.transitionMatrix.type());
+
+        // initial state
+        state->statePost = kfParams.state;
+        state->errorCovPost = kfParams.errorCov;
+
+        // dynamic system initialization
+        state->controlMatrix = kfParams.controlMatrix;
+        state->measurementMatrix = kfParams.measurementMatrix;
+        state->transitionMatrix = kfParams.transitionMatrix;
+        state->processNoiseCov = kfParams.processNoiseCov;
+        state->measurementNoiseCov = kfParams.measurementNoiseCov;
+    }
+
+    static void run(const cv::Mat& measurements, bool haveMeasurement,
+                    const cv::Mat& control, const cv::gapi::KalmanParams&,
+                    cv::Mat &out, cv::KalmanFilter& state)
+    {
+        cv::Mat pre = cv::norm(control, cv::NORM_INF) != 0 ? state.predict(control) : state.predict();
+        haveMeasurement ? state.correct(measurements).copyTo(out) : pre.copyTo(out);
+    }
+};
+
+GAPI_OCV_KERNEL_ST(GCPUKalmanFilterNoControl, cv::gapi::video::GKalmanFilterNoControl, cv::KalmanFilter)
+{
+    static void setup(const cv::GMatDesc&, const cv::GOpaqueDesc&,
+                      const cv::gapi::KalmanParams& kfParams,
+                      std::shared_ptr<cv::KalmanFilter> &state,
+                      const cv::GCompileArgs&)
+    {
+        state = std::make_shared<cv::KalmanFilter>(kfParams.transitionMatrix.rows, kfParams.measurementMatrix.rows,
+                                                   0, kfParams.transitionMatrix.type());
+        // initial state
+        state->statePost = kfParams.state;
+        state->errorCovPost = kfParams.errorCov;
+
+        // dynamic system initialization
+        state->measurementMatrix = kfParams.measurementMatrix;
+        state->transitionMatrix = kfParams.transitionMatrix;
+        state->processNoiseCov = kfParams.processNoiseCov;
+        state->measurementNoiseCov = kfParams.measurementNoiseCov;
+    }
+
+    static void run(const cv::Mat& measurements, bool haveMeasurement,
+                    const cv::gapi::KalmanParams&, cv::Mat &out,
+                    cv::KalmanFilter& state)
+    {
+        cv::Mat pre = state.predict();
+        haveMeasurement ? state.correct(measurements).copyTo(out) : pre.copyTo(out);
```

Comment:
Prefer to use if/else and dedicated lines for statements in such cases (it is critical for accurate code coverage reports)

(here and above)

Reply:
Done.